### ISSUE TITLE
Utilizing Menus in Quorum Studio 3.0 with JAWS

### ISSUE THREAD

**Issue by DarkhorseStudios** [qorf/quorum-studio#1]
Hey all,
I'm using Quorum Studio 3.0 with JAWS 2020, and am unable to use menus/hotkeys inside of Quorum studio to run my code, debug, etc. It's standard in accessible apps to be able to access and utilize the menus in two ways with JAWS:
1: Pressing the specified key combinations, such as control + R to run a project.
However, when I do this, JAWS announces that I've pressed control and R, but nothing happens; the code editor area remains as it was, and no additional windows open inside of Quorum Studio. Pressing control + tab does cycle through any other code editor windows that I have open, but that's it.

2: navigating to the menus with the alt key, then traversing the menus with the arrows keys. For this, using the up/down arrow keys allows you to interact with a menu, and its' first layer of options. Then when JAWS is focused on the desired option, you can typically use either the enter keys or the space bar to select that option. 

For example, I can press alt to bring my focus to the menu bar, with the focus initially settling on the "File menu. I can then move the focus laterally between the menus, to the "Run" menu, for example. From here, pressing the down arrow key works correctly, and opens up a number of new options for me to access. When highlighting one of the submenus, like the "run" submenu, pressing either of the enter keys or the space bar just returns JAWS' focus to the code editor area, with no changes occurring. Additionally, when attempting to interact with the submenus using the right arrow, the next level of options presented is the same whether I interact with the build, clean, build and clean, or run submenus:

clone, pull, status, add control, add proje t files, commit, and push. 
Again, pressing the enter keys or the space bar to select any of these options has the same result.

As I'm still new to programming, I don't have the skills or knowledge required to take a shot at fixing this, and was hoping someone else can take a crack at it. Thank you!

**Comment by andreas-stefik** [comment]
We will take a look. When I press those keys, it does actually interact with Quorum Studio. Can you please confirm which version of JAWS and Windows you are using? Older versions of JAWS sometimes don't have compatibility with Microsoft's UIA architecture, which is what Quorum Studio uses for Accessibility support. It has been improving in newer versions though.

That said, pressing enter or alt on the menus did actually work for me just now, so let me know what version of JAWS you are on. I'll see if we can't get a copy and check. Do you get the same behavior in NVDA? It tends to be more standards compliant.

**Comment by andreas-stefik** [comment]
DarkhorseStudios,

I was able to grab a machine with JAWS 2020, which I think on the mailing list you said is the one you are using. In that version, I tried:

1. Going to the menus and pressing ENTER or SPACE.
2. Opening a program and pressing CTRL + R

I have a hypothesis as to what is going on here, although I'm not sure so please tell me if I'm wrong. All of the items you mention are following the same pattern and I think it is a usability issue, but not really a bug. 

When I open something like Notepad in JAWS and press the same keys, if that key opens a dialog or something similar, jaws announces it. However, in Quorum Studio, there are some actions that dump information into the console and this can be a small amount of information or a lot, but it doesn't hook this up by default to an ARIA-LIVE style region (to use terminology from the web, even though it's different in Quorum), because this could be overwhelming.

For example, suppose I have the program:

output "Yo"

If I then Open Quorum Studio and press CTRL + R, it really does appear like it does nothing, but that's not a bug. That's because my the program I wrote was specifically designed to output something to the console, but nothing else. If I want to access what my program sent to the console, I can press CTRL + 3 and it will take me to the console, where it has the following output in a project I called Howdy:

Building Howdy
Running Howdy
Build Successful in 0.20 seconds
Yo

The same goes for push/pull etc. Those commands work, but they dump info to the console in the same way. Sometimes they dump quite a bit of information, depending on what you are doing. A commit could be 1 file or thousands, for example. If instead I wrote a program that pops up a window, that window will pop up and announce itself as a new program you wrote.

I feel like you're right though. That isn't very intuitive, even if I am used to it and a common pattern in menus. What would you prefer it does for each of these?

Also, thank you for bringing this up. I have really appreciated all the messages you have sent to the message and look forward to learning more about how you are using everything as you go.

**Comment by andreas-stefik** [comment]
One more thought here before I jet. I updated my license on this machine to JAWS 2021 and it does seem like there have been improvements and things are working better, even though the Quorum code is the same. For example, in JAWS 2021, when I press CTRL +R, with the same program, it now announces "Build Successful in 0.04 seconds," as opposed to forcing you to go to the console, so it must be picking up better on our UI Automation hooks now. That's great, as that is what I am used to in NVDA.
